### DISCUSSION TITLE
Swap display contents alltogether

### DISCUSSION THREAD

**Discussion by kubudi** [rxhanson/Rectangle#819] [Ideas]
Hello,

I have a specific setup for my displays that made me wonder if this feature might be helpful for me and people with a similar form.

The way I use my displays is uneven. One of them is slightly more centered, which gets my main focus. The other display is slightly leaning toward the edge of the table. 

While I code, the editor has the main focus, and documentation is on my side screen.
While I search for something, I keep the editor on the side display for reference. 

As you might have guessed, I often switch between these two usages.
Thus, I found myself swapping the contents of my displays manually to proceed with what I do.
Editor to the left, browser to the right. And vice versa.

What I'm wondering is, is this feature even possible to do with Rectangle. Since you need to move not only the window on focus, you also need access to the defocused window on the other display.

TL;DR:

- Do you think this feature might prove helpful? Would you use it?
- Is it even possible to implement?

I would be more than happy to work on this if it comes down to that.
P.S. I looked around but couldn't find a similar discussion. Please point me there if this has been discussed before. 

Thanks <3

**Comment by rxhanson** [comment]
Makes sense to me, I see no reason to turn down a good pull request. This is something that I won't let into the UI at first, but can see adding it in later if there is significant demand. 

Here's a similar feature for reversing all the windows on a single display: https://github.com/rxhanson/Rectangle/commit/673e903cf3fdf9615aa400f12f23073c7f8c26a0

It's from this PR: https://github.com/rxhanson/Rectangle/pull/526

Adding it that way makes it configurable via terminal command, out of the box.

**Reply by kubudi** [reply]
Okay, let me play around a bit then 👍

**Reply by bjornorri** [reply]
Hi kubudi, any news on this?

I have a similar feature request for swapping the topmost windows on the displays. This could be generalized as cycling the topmost windows between displays (if displays > 2).

I'd be happy to take a stab at this myself, but would rather avoid duplicate work.

**Reply by kubudi** [reply]
Not yet, unfortunately. Feel free to jump in!

**Comment by andrecarlucci** [comment]
I would definitely use this!
